source/com/github/tombentley/alabama/Deserializer.ceylon

```diff
@@ -554,21 +554,35 @@ Class<Object> bestType(Type<> modelType, Type<> keyType) {
 }
 
 "Given a Type reflecting an Iterable, returns a Type reflecting the 
- iterated type or returns null if the given Type does not reflect an Iterable"
+ iterated type or returns null if the given Type does not reflect an Iterable.
+ If Type is UnionType, preferredTypeIfUnion argument is needed to determine preferredType
+ "
 by("jvasileff")
-Type<Anything> iteratedType(Type<Anything> containerType) {
+Type<Anything> iteratedType(Type<Anything> containerType, Type<Anything> preferredTypeIfUnion = `Anything`) {
+    return iteratedTypeOrNull(containerType, preferredTypeIfUnion) else `Nothing`;
+}
+
+Type<Anything>? iteratedTypeOrNull(Type<Anything> containerType, Type<Anything> preferredTypeIfUnion) {
     if (is ClassOrInterface<Anything> containerType) {
         if (exists model = containerType.satisfiedTypes
-                .narrow<Interface<Iterable<Anything>>>().first,
+            .narrow<Interface<Iterable<Anything>>>().first,
             exists x = model.typeArgumentList.first) {
             //print("iteratedType(containerType=``containerType``): ``x``");
             return x;
         } else if (containerType.supertypeOf(`Iterable<Anything>`)) {
             return `Anything`;
         }
     }
-    
-    return `Nothing`;
+    if (is UnionType<> containerType) {
```

Comment:
The method's name says suggests it's about iterable/iterated types, but you've bolted this special case for union types onto the side of it. What's so important about union types that they deserve this special treatment in this method? 

Since you only call the two argument version of `iteratedType()` in one place wouldn't it be better to handle the special case there, at the call site, rather than changing `iteratedType()` to add this special case?